LCM/scripts/RemoveModule.py

```diff
@@ -4,124 +4,187 @@
-            RemoveMirroredDirectory(moduleDir + "/" + curFile, omiDir + "/" + curFile)
-            os.rmdir(moduleDir + "/" + curFile)
-            if len(os.listdir(omiDir + "/" + curFile)) == 0:
-                os.rmdir(omiDir + "/" + curFile)
-
-        if os.path.islink(omiDir + "/" + curFile):
-            os.path.unlink(omiDir + "/" + curFile)
-        elif os.path.isfile(omiDir + "/" + curFile):
-            os.remove(omiDir + "/" + curFile)
-
-if len(sys.argv) != 2:
-    usage()
-
-moduleName = sys.argv[1]
-omi_bindir = "<CONFIG_BINDIR>"
-omi_libdir = "<CONFIG_LIBDIR>"
-omi_sysconfdir = "<CONFIG_SYSCONFDIR>"
-baseModulePath = "<DSC_MODULES_PATH>"
-baseScriptPath = "<DSC_SCRIPT_PATH>"
-modulePath = baseModulePath + "/" + moduleName
-omiregister_dir = "/etc/opt/omi/conf/omiregister"
-
-if not os.path.isdir(modulePath):
-    print("Error: unable to find installed module " + moduleName + " at " + modulePath)
-    sys.exit(1)
+def printVerboseMessage(message):
+    verboseMessage = "VERBOSE from RemoveModule.py: " + message
+    print(verboseMessage)
 
-print("Removing module " + moduleName)
+def getPlatformArchitectureFolderName():
+    platformArchitecture = platform.architecture()
+    if len(platformArchitecture) != 2:
+        exitWithError("The python function platform.architecture() failed to return a valid tuple. Cannot detect if this system has an x64 or an x86 architecture.")
 
-# special section for nxOMSAutomationWorker module
-# the Linux Hybrid worker and manager needs to be killed before the module is removed
-# also a good idea to remove the state and working directory
-if moduleName == "nxOMSAutomationWorker":
-    state_dir_path = "/var/opt/microsoft/omsagent/state/automationworker"
-    working_dir_path = "/var/opt/microsoft/omsagent/run/automationworker"
-    automation_user = "nxautomation"
+    if (platformArchitecture[0] == "64bit"):
+        platformArchitectureFolderName = "x64"
+    else:
+        platformArchitectureFolderName = "x86"
 
-    # invoke deregister when it becomes available
-    # Kill all processes running as nxautomation
-    # In some cases, there might be no processes to kill, and the return code of the following command might be non-zero.
-    # We make a best attempt to terminate the processes and don't care about its return code
-    subprocess.call(["sudo", "pkill", "-u", automation_user, ".*"])
+    return platformArchitectureFolderName
 
-    # remove directories
-    shutil.rmtree(state_dir_path, ignore_errors=True)
-    shutil.rmtree(working_dir_path, ignore_errors=True)
+def regenerateDscPythonScriptInitFiles():
+    regenerateInitFilesScriptPath = join(helperlib.DSC_SCRIPT_PATH, 'RegenerateInitFiles.py')
+    regenerateInitFilesResult = subprocess.call(regenerateInitFilesScriptPath)
+    if regenerateInitFilesResult != 0:
+        exitWithError("Failed to regenerate the DSC __init__.py files with the result code " + regenerateInitFilesResult)
```

Comment:
This function re-writes the __init__.py files that define what resource Python script files DSC imports (e.g. MSFT_nxPackage.py). Any time a resource script is added or removed, the same function needs to run to update the file to include or remove the resource's script from the list in the __init__.py file. This same call was in both files before my change as well. I just moved it to a function to make it easier to move the function to a library later.

Ultimately, I would like to have a common library between these two scripts, but I don't want to add any more un-needed complexity to this change by adding a new Python library structure since this fix is needed quickly and urgently.